### PR TITLE
Add SeekStride to pool configuration

### PR DESCRIPTION
**Pull request by mattnibs**
Instead of having SeekStride as a global variable, move this to an
option in pools.Config. Drop -seekstride as a flag in zed command load
and add -seekstride as a hidden flag in zed create.

### CODE REVIEW

File: lake/data/object.go [hunk 1]
```diff
@@ -14,7 +14,10 @@ import (
 	"github.com/segmentio/ksuid"
 )
 
-const DefaultThreshold = 500 * 1024 * 1024
+const (
+	DefaultThreshold  = 500 * 1024 * 1024
+	DefaultSeekStride = 64 * 1024
```
Comment (nwt) [inline lake/data/object.go:19]:
```suggestion
	DefaultSeekStride = 64 * 1024
	DefaultThreshold  = 500 * 1024 * 1024
```

File: api/api.go [hunk 2]
```diff
@@ -36,9 +36,10 @@ type VersionResponse struct {
 }
 
 type PoolPostRequest struct {
-	Name   string       `json:"name"`
-	Layout order.Layout `json:"layout"`
-	Thresh int64        `json:"thresh"`
+	Name       string       `json:"name"`
+	Layout     order.Layout `json:"layout"`
+	Thresh     int64        `json:"thresh"`
+	SeekStride int          `json:"seek_stride"`
```
Comment (nwt) [inline api/api.go:42]:
```suggestion
	SeekStride int          `json:"seek_stride"`
	Thresh     int64        `json:"thresh"`
```

File: lake/api/remote.go [hunk 3]
```diff
@@ -47,11 +47,12 @@ func (r *RemoteSession) CommitObject(ctx context.Context, poolID ksuid.KSUID, br
 	return res.Commit, err
 }
 
-func (r *RemoteSession) CreatePool(ctx context.Context, name string, layout order.Layout, thresh int64) (ksuid.KSUID, error) {
+func (r *RemoteSession) CreatePool(ctx context.Context, name string, layout order.Layout, thresh int64, seekStride int) (ksuid.KSUID, error) {
```
Comment (nwt) [inline lake/api/remote.go:50]: Maybe keep the order of these consistent with that of the api.PoolPostRequest fields?
```suggestion
func (r *RemoteSession) CreatePool(ctx context.Context, name string, layout order.Layout, seekStride int, thresh int64) (ksuid.KSUID, error) {
```

**Review by nwt — APPROVED** [review]

File: lake/data/object.go (hunk 1 above)
Comment (mattnibs) [reply lake/data/object.go:19]: Doh!

**Pull request merged**